packages/react-utilities/src/utils/isFluentTriggerComponent.ts

```diff
@@ -0,0 +1,26 @@
+import * as React from 'react';
+
+/**
+ * Allows a component to be tagged as a FluentUI trigger component.
+ *
+ * Triggers are special-case components: they attach event listeners and other props on their child,
+ * and use them to trigger another component to show. Examples include `MenuTrigger` and `Tooltip`.
+ *
+ * A component can be tagged as a trigger as follows:
+ * ```ts
+ * const MyComponent: React.FC<MyComponentProps> & FluentTriggerComponent = ...;
+ *
+ * MyComponent.isFluentTriggerComponent = true; // MUST also set this to true
+ * ```
+ */
+export type FluentTriggerComponent = {
+  isFluentTriggerComponent?: boolean;
+};
+
+/**
+ * Checks if a given component type is a FluentUI trigger (e.g. `MenuTrigger` or `Tooltip`).
+ * See the {@link FluentTriggerComponent} type for more info.
+ */
+export const isFluentTriggerComponent = <P>(type: React.ReactElement<P>['type'] & FluentTriggerComponent) => {
+  return typeof type !== 'string' && (type as FluentTriggerComponent).isFluentTriggerComponent;
+};
```

Comment:
```suggestion
export const isFluentTriggerElement = <P>(type: React.ReactElement & FluentTriggerComponent) => {
  return (child.type as FluentTriggerComponent).isFluentTriggerComponent;
};
```

Should not it be just `isFluentTriggerElement()`? Is there sense to check that `type` is `string`?